paimon-flink/paimon-flink-common/src/test/java/org/apache/paimon/flink/StreamingWarehouseITCase.java

```diff
@@ -214,6 +228,157 @@ public void testUserStory() throws Exception {
         dailyTaskHandler.cancel().get();
     }
 
+    @ParameterizedTest
+    @EnumSource(MergeEngine.class)
+    public void testUpdateWithPrimaryKey(MergeEngine mergeEngine) throws Exception {
```

Comment:
I prefer to place this test in `ReadWriteTableITCase` for 2 reason:
1. Many SQL-related tests are there. You can put tests in `Keyword` field or a new field like `Update statement`. 
2. This test class seems a dedicate class for a whole pipeline.